plugin/documents.py

```diff
@@ -102,6 +112,15 @@ def on_activated_async(self) -> None:
         if self._is_regular_view() and not self.view.is_loading():
             self._register_async()
 
+    def on_modified_async(self) -> None:
+        sel = self.view.sel()
+        if len(sel) < 1:
+            return
+        current_point = sel[0].b
+        if self._stored_point != current_point:
+            self._stored_point = current_point
+            debounced(self._do_document_color, 800, lambda: self._stored_point == current_point, async_thread=True)
```

Comment:
Can we avoid debouncing entirely when colorProvider is disabled (even if we have to check settings twice)?

Reply:
Done in 0a0bd9b